src/services/blocks/BlocksService.ts

```diff
@@ -558,9 +558,10 @@ export class BlocksService extends AbstractService {
 		blockHash: BlockHash
 	): Promise<WeightValue> {
 		const metadata = await api.rpc.state.getMetadata(blockHash);
+		const historicApi = (await api.at(blockHash)) as ApiPromise;
```

Comment:
@TarikGul I suppose the registry contains metadata, chain info and other stuff that is required for that data to match for historical blocks?

Reply:
Yup exactly! 

Example: The specific error that brought up this issue too was a type error where Westend has a `Lookup36` type in its Metadata, but it didnt match the registry. So when trying to expand the metadata, types didn't align.